### PR TITLE
[YUNIKORN-815] Fix scheduler interface makefile issues.

### PR DESCRIPTION
**Pull request by yangwwei**
### What is this PR for?
After https://issues.apache.org/jira/browse/YUNIKORN-760, we are seeing issues updating the scheduler-interface dependency in other repos. This is because #40 has removed the GRPC generated code, but we have useful code in the core repo that still needs them, such as https://github.com/apache/incubator-yunikorn-core/tree/master/cmd.

There is another issue that has been fixed in this PR, #46 added a constant, that should go into the md file instead of the source code; the source code file is also generated. cc @HuangTing-Yao .

This PR also does more proper cleanup for `make clean`. Earlier if we have old downloaded protobuf binaries, we are skipping to download new files, which could cause problems to build files with old protobuf stuff. To avoid that, we should suggest to run `make clean` first for proper cleanup.

### What type of PR is it?
* [x] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
Update the scheduler-core and shim repo dependencies and let them depend on the latest scheduler-interface.

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-815

### How should this be tested?
Updated dep in my local workspace to make sure nothing is broken.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by yangwwei** [comment]
@wilfred-s , @HuangTing-Yao  pls help to review, thx

**Comment by HuangTing-Yao** [comment]
Hi @wilfred-s , I don't familiar with interface code, could you help to review this ?
Thanks.

File: scheduler-interface-spec.md [hunk 1]
```diff
@@ -673,8 +673,9 @@ const (
 	GroupAnnotation = "annotation/"
 
 	// Keys
-	KeyPodName   = "podName"
-	KeyNamespace = "namespace"
+	KeyPodName      = "podName"
+	KeyNamespace    = "namespace"
+	KeyRequiredNode = "requiredNode"
```
Comment (kingamarton) [inline scheduler-interface-spec.md:678]: Is this related to build fix? If it is not I think we should address in a different Jira/PR.

File: go.mod [hunk 2]
```diff
@@ -19,10 +19,10 @@
 
 module github.com/apache/incubator-yunikorn-scheduler-interface
 
-go 1.12
+go 1.15
```
Comment (kingamarton) [inline go.mod:22]: I think for updating the go version is better to create a different Jira, it makes easyer to trach the changes, since this issue is a bout a build failure, not about go version upgrade.

**Review by kingamarton — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (yangwwei) [reply go.mod]: Sure, will address this in a follow-up JIRA

File: scheduler-interface-spec.md (hunk 1 above)
Comment (yangwwei) [reply scheduler-interface-spec.md]: this is not required for this PR
I've reopened https://issues.apache.org/jira/browse/YUNIKORN-810 to fix this.

**Review by kingamarton — APPROVED** [review]
